frontend/www/js/omegaup/components/course/StudentProgress.vue

```diff
@@ -120,6 +121,27 @@ export default class StudentProgress extends Vue {
     );
   }
 
+  get weightedMean(): Array<number> {
+    const totalPoints =
+      this.assignments
+        .map((assignment) => assignment.max_points)
+        .reduce((a, b) => (a ?? 0) + (b ?? 0)) ?? 0;
+    return this.assignments.map(
+      (assignment) => ((assignment.max_points ?? 0) / (totalPoints ?? 0)) * 100,
```

Comment:
```suggestion
      (assignment) => totalPoints ? ((assignment.max_points ?? 0) / totalPoints) * 100 : 0,
```

`totalPoints` no puede ser `undefined` o `null` bajo ninguna circunstancia, así que no es necesario el `?? 0`. pero `totalPoints` sí puede ser 0, y eso causaría problemas.

Reply:
Listo